12-watchtower-API.md

```diff
@@ -270,73 +363,31 @@ Sample code (python) for the client to prepare the `encrypted_blob`:
+Although this BOLT does not enforce any specific payment method to be adopted, it is worth mentioning the three most common ones:
 
 **On-chain bounty**. An additional output is created in the justice transaction that will reward the WatchTower. 
 
-**Micropayments**. A small payment is sent to the WatchTower for every new job (i.e. over the lightning network)
-
-**Subscription**. WatchTower is periodically rewarded / paid for their service to the customer. (i.e. over the lightning network or dirty fiat subscription). 
+**Micropayments**. A small payment is sent to the WatchTower for every new job (e.g. over the lightning network)
 
-Both micropayments and subscriptions are favourable for a WatchTower.
+**Subscription**. WatchTower is periodically rewarded / paid for their service to the customer. (e.g. over the lightning network or fiat subscription). 
 
-**Problems with onchain bounty** We highlight that the on-chain bounty approach is not ideal for a watching network. It lets the customer hire N WatchTowers (O(N) storage for each tower) and only one WatchTower will be rewarded upon collecting the bounty. If this approach was sought, then the miners may end up becoming WatchTowers as they can very easily front-run an entire watching network. Thus to reduce the power and responsibility of miners, we should avoid this approach and advocate that miners do not participate in a watching network. 
- 
-## Number of updates
+Both micropayments and subscriptions are favourable for a WatchTower. The on-chain bounty approach is not ideal for a watching network, it lets the customer hire N WatchTowers (O(N) storage for each tower) and only one WatchTower will be rewarded upon collecting the bounty. On top of that, the onchain bounty allows a network-wise DoS attack for free.
 
-To offer full protection, a WatchTower requires an encrypted blob for every single update in the channel. This is a symptom of replace-by-revocation channels as there is a single valid state and a set of "revoked" states. Only one of the states can be broadcast and this is up to the counterparty (cheater) to decide. 
+## No compression of justice transaction 
 
-All signed receipts only correspond to a single job / justice transaction. Thus the customer should keep a copy of all signed receipts received from the WatchTower. A future BOLT can extend the signed receipt to include a committment to all previous encrypted blobs (merkle tree) to reduce this storage. 
+The storage requirements for a WatchTower can be reduced (linearly) by implementing [shachain](https://github.com/rustyrussell/ccan/blob/master/ccan/crypto/shachain/design.txt), therefore storing the parts required to build the transaction and the corresponding signing key instead of the full transaction. For now, we have decided to keep the hiring protocol simple. Storage is relatively cheap and we can revisit this standard if it becomes a problem. 
 
-## No compression of justice transaction 
+## FIXMES
```

Comment:
Is FIXME a BOLT thing? Or do all of these need removing before you submit?

Reply:
This are pending things that I'll like to discuss/add in later iterations.

Some BOLTs have fixmes in the middle.